wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,202 @@
+- Introduce enhancements across multiple Kubernetes SIGs to add support and integration for the new
+  APIs to solve wide range of issues.
+
+We expect to provide reference implementations of the new APIs including but not limited to
+controllers, API validation, integration with existing core components and extension points for the
+ecosystem. This should be accompanied by E2E / Conformance tests.
+
+## Relevant Features, KEPs and Documents
+
+- Declarative Node Maintenance: https://github.com/kubernetes/enhancements/issues/4212
+- EvictionRequest API: https://github.com/kubernetes/enhancements/issues/4563
+- Graceful Node Shutdown [Beta]: https://github.com/kubernetes/enhancements/issues/2000
+- DRA: device taints and tolerations [Alpha]: https://github.com/kubernetes/enhancements/issues/5055
+- Disrupted Pods should be removed from endpoints: https://docs.google.com/document/d/1t25jgO_-LRHhjRXf4KJ5xY_t8BZYdapv7MDAxVGY6R8
+- Node Readiness Gates: https://github.com/kubernetes/enhancements/issues/5233
+- Allow the kubelet to trigger rescheduling of pods: https://docs.google.com/document/d/1-wJhiNy84w7tzFdo9HqwTu5DrVSuXFLGTUv8FBiRAAc
+- Implicit tolerations https://github.com/kubernetes/enhancements/issues/5282; add a Filter plugin to ensure that non-GPU pods are not scheduled on GPU nodes: https://github.com/kubernetes-sigs/scheduler-plugins/pull/812
+- Node Resource Hot Plug: https://github.com/kubernetes/enhancements/issues/3953
+
+## Relevant Projects
+
+This is a list of known projects that solve similar problems in the ecosystem or would benefit from
+the efforts of this WG:
+
+- https://github.com/aws/aws-node-termination-handler
+- https://github.com/foriequal0/pod-graceful-drain
+- https://github.com/jukie/karpenter-deprovision-controller
+- https://github.com/kubereboot/kured
+- https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
+- https://github.com/kubernetes-sigs/cluster-api/
+- https://github.com/kubernetes-sigs/karpenter
+- https://github.com/kubernetes-sigs/kubespray
+- https://github.com/kubevirt/kubevirt
+- https://github.com/medik8s/node-maintenance-operator
+- https://github.com/Mellanox/maintenance-operator
+- https://github.com/NVIDIA/pika
+- https://github.com/openshift/machine-config-operator
+- https://github.com/planetlabs/draino
+- https://github.com/strimzi/drain-cleaner
+
+There are also internal custom solutions that companies use.
+
+## Prioritization
+
+The group activity will focus on the following features:
+
+- Declarative Node Maintenance
+- EvictionRequest API
+- Graceful Node Shutdown
+
+And have the following priorities which mostly apply to WG calls, but can also apply to the general
+WG review/work/guidance capacity:
+1. Urgent topics concerning the WG focus features during the KEP and code greeze periods.
+2. Discussing issues within the scope of the WG.
+3. Presentations within the scope of the WG.
+4. The WG focus features outside the KEP and code freeze periods.
```

Comment:
Drop this point, it mostly expresses the same thing what the next one, only the latter uses better wording. This one suggests, the group will work outside of general release timeline, which is not allowed.

Reply:
done